### DISCUSSION TITLE
2.0版用gradle 6.8编译失败

### DISCUSSION THREAD

**Discussion by andersgong** [mamoe/mirai#840] [Q&A] [answered]
我的环境是jdk-15.0.1加gradle-6.8，请看下为什么会报这个错误呢？不论用M2分支还是dev分支都一样，谢谢。

> Configure project :mirai-core-all
e: D:\mirai_official\mirai-core-all\build.gradle.kts:30:1: Unresolved reference: configurePublishing

FAILURE: Build failed with an exception.

* Where:
Build file 'D:\mirai_official\mirai-core-all\build.gradle.kts' line: 30

* What went wrong:
Script compilation error:

  Line 30: configurePublishing("mirai-core-all")
           ^ Unresolved reference: configurePublishing

1 error

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 15s

**Comment by Him188** [comment, marked as answer]
Gradle 6.8 更改了 Kotlin script 编译, 尝试 `./gradlew clean` 再 `./gradlew build`

**Reply by andersgong** [reply]
好的，我去试一下，感谢回答。另外，我想问下，建议使用dev还是2.0-M2-1-release呢？

**Reply by Him188** [reply]
dev 分支会包含所有开发提交.
2.0-M2-1-release 分支是从 dev 里筛选一些兼容的变更用来发 2.0-M2 的 bugfix (patch)版本. 

因此如果你要为 mirai 提交 PR, 请基于 dev. 如果只需要使用 mirai, 在 [release](https://github.com/mamoe/mirai/releases) 选择最新版本并参考 [ConfiguringProjects.md](https://github.com/mamoe/mirai/blob/dev/docs/ConfiguringProjects.md)

**Comment by Him188** [comment]
mirai dev 分支我更新到 Gradle 6.8 了, 测试可以编译

**Reply by andersgong** [reply]
威武，我正在试6.7，下载那些包有点慢，等我测试好再试试6.8，谢谢。

**Reply by andersgong** [reply]
晕，还是不行，我把C:\用户下面的.gradle目录删掉，重新开始，结果还是一样，下面是整个过程，我看了是6.8：

D:\mirai_official>gradlew build
Starting a Gradle Daemon (subsequent builds will be faster)

> Configure project :mirai-core
Kotlin Multiplatform Projects are an Alpha feature. See: https://kotlinlang.org/docs/reference/evolution/components-stability.html. To hide this message, add 'kotlin.mpp.stability.nowarn=true' to the Gradle properties.

Android SDK 可能未安装.
                mirai-core 的 Android 目标编译将不会进行.
                这不会影响 Android 以外的平台的编译.
Android SDK might not be installed.
                Android target of mirai-core will not be compiled.
                It does no influence on the compilation of other platforms.
configuration ':mirai-core:allSourceSetsCompileDependenciesMetadata', configuration ':mirai-core:allSourceSetsRuntimeDependenciesMetadata', configuration ':mirai-core:annotationProcessor', configuration ':mirai-core:apiElements', configuration ':mirai-core:archives', configuration ':mirai-core:commonApi', configuration ':mirai-core:commonApiElements', configuration ':mirai-core:commonCompile', configuration ':mirai-core:commonCompileClasspath', configuration ':mirai-core:commonCompileOnly', configuration ':mirai-core:commonDefault', configuration ':mirai-core:commonImplementation', configuration ':mirai-core:commonMainApi', configuration ':mirai-core:commonMainApiDependenciesMetadata', configuration ':mirai-core:commonMainCompileOnly', configuration ':mirai-core:commonMainCompileOnlyDependenciesMetadata', configuration ':mirai-core:commonMainImplementation', configuration ':mirai-core:commonMainImplementationDependenciesMetadata', configuration ':mirai-core:commonMainMetadataElements', configuration ':mirai-core:commonMainRuntimeOnly', configuration ':mirai-core:commonMainRuntimeOnlyDependenciesMetadata', configuration ':mirai-core:commonRuntime', configuration ':mirai-core:commonRuntimeClasspath', configuration ':mirai-core:commonRuntimeElements', configuration ':mirai-core:commonRuntimeOnly', configuration ':mirai-core:commonSourceArtifacts', configuration ':mirai-core:commonTestApi', configuration ':mirai-core:commonTestApiDependenciesMetadata', configuration ':mirai-core:commonTestCompile', configuration ':mirai-core:commonTestCompileClasspath', configuration ':mirai-core:commonTestCompileOnly', configuration ':mirai-core:commonTestCompileOnlyDependenciesMetadata', configuration ':mirai-core:commonTestImplementation', configuration ':mirai-core:commonTestImplementationDependenciesMetadata', configuration ':mirai-core:commonTestRuntime', configuration ':mirai-core:commonTestRuntimeClasspath', configuration ':mirai-core:commonTestRuntimeOnly', configuration ':mirai-core:commonTestRuntimeOnlyDependenciesMetadata', configuration ':mirai-core:compile', configuration ':mirai-core:compileClasspath', configuration ':mirai-core:compileOnly', configuration ':mirai-core:default', configuration ':mirai-core:implementation', configuration ':mirai-core:jvmApi', configuration ':mirai-core:jvmApiElements', configuration ':mirai-core:jvmCompile', configuration ':mirai-core:jvmCompileClasspath', configuration ':mirai-core:jvmCompileOnly', configuration ':mirai-core:jvmDefault', configuration ':mirai-core:jvmImplementation', configuration ':mirai-core:jvmMainApi', configuration ':mirai-core:jvmMainApiDependenciesMetadata', configuration ':mirai-core:jvmMainCompileOnly', configuration ':mirai-core:jvmMainCompileOnlyDependenciesMetadata', configuration ':mirai-core:jvmMainImplementation', configuration ':mirai-core:jvmMainImplementationDependenciesMetadata', configuration ':mirai-core:jvmMainRuntimeOnly', configuration ':mirai-core:jvmMainRuntimeOnlyDependenciesMetadata', configuration ':mirai-core:jvmRuntime', configuration ':mirai-core:jvmRuntimeClasspath', configuration ':mirai-core:jvmRuntimeElements', configuration ':mirai-core:jvmRuntimeOnly', configuration ':mirai-core:jvmSourceArtifacts', configuration ':mirai-core:jvmTestApi', configuration ':mirai-core:jvmTestApiDependenciesMetadata', configuration ':mirai-core:jvmTestCompile', configuration ':mirai-core:jvmTestCompileClasspath', configuration ':mirai-core:jvmTestCompileOnly', configuration ':mirai-core:jvmTestCompileOnlyDependenciesMetadata', configuration ':mirai-core:jvmTestImplementation', configuration ':mirai-core:jvmTestImplementationDependenciesMetadata', configuration ':mirai-core:jvmTestRuntime', configuration ':mirai-core:jvmTestRuntimeClasspath', configuration ':mirai-core:jvmTestRuntimeOnly', configuration ':mirai-core:jvmTestRuntimeOnlyDependenciesMetadata', configuration ':mirai-core:kotlinCompilerClasspath', configuration ':mirai-core:kotlinCompilerPluginClasspath', configuration ':mirai-core:kotlinKlibCommonizerClasspath', configuration ':mirai-core:kotlinNativeCompilerPluginClasspath', configuration ':mirai-core:kotlinScriptDef', configuration ':mirai-core:kotlinScriptDefExtensions', configuration ':mirai-core:metadataApi', configuration ':mirai-core:metadataApiElements', configuration ':mirai-core:metadataCompile', configuration ':mirai-core:metadataCompileClasspath', configuration ':mirai-core:metadataCompileOnly', configuration ':mirai-core:metadataDefault', configuration ':mirai-core:metadataImplementation', configuration ':mirai-core:metadataRuntimeOnly', configuration ':mirai-core:runtime', configuration ':mirai-core:runtimeClasspath', configuration ':mirai-core:runtimeElements', configuration ':mirai-core:runtimeOnly', configuration ':mirai-core:testAnnotationProcessor', configuration ':mirai-core:testCompile', configuration ':mirai-core:testCompileClasspath', configuration ':mirai-core:testCompileOnly', configuration ':mirai-core:testImplementation', configuration ':mirai-core:testKotlinScriptDef', configuration ':mirai-core:testKotlinScriptDefExtensions', configuration ':mirai-core:testRuntime', configuration ':mirai-core:testRuntimeClasspath', configuration ':mirai-core:testRuntimeOnly'

> Configure project :mirai-core-all
e: D:\mirai_official\mirai-core-all\build.gradle.kts:30:1: Unresolved reference: configurePublishing

FAILURE: Build failed with an exception.

* Where:
Build file 'D:\mirai_official\mirai-core-all\build.gradle.kts' line: 30

* What went wrong:
Script compilation error:

  Line 30: configurePublishing("mirai-core-all")
           ^ Unresolved reference: configurePublishing

1 error

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 6m 33s

D:\mirai_official>

**Reply by Him188** [reply]
尝试重新 clone

**Reply by andersgong** [reply]
我是直接clone一个新目录，然后就在那个目录里执行gradlew build，而且6.7也这样，实在是找不到原因了。

configurePublishing("mirai-core-all")
就是这一句，怎么都过不去，怪事。

**Reply by andersgong** [reply]
> 
> 
> 尝试重新 clone

不好意思，我的锅，6.8编译成功，大佬威武，非常感谢！
